spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/KubernetesNamespaceProvider.java

```diff
@@ -22,16 +22,16 @@
 import java.nio.file.Paths;
 
 import org.springframework.boot.logging.DeferredLog;
-import org.springframework.context.ApplicationEvent;
-import org.springframework.context.ApplicationListener;
 import org.springframework.core.env.Environment;
 
 import static org.springframework.cloud.kubernetes.commons.KubernetesClientProperties.SERVICE_ACCOUNT_NAMESPACE_PATH;
 
 /**
  * @author Ryan Baxter
  */
-public class KubernetesNamespaceProvider implements ApplicationListener<ApplicationEvent> {
```

Comment:
I removed `implements ApplicationListener<ApplicationEvent>` and moved the `LOG.replayTo(KubernetesNamespaceProvider.class);` into the constructor. I am actually not sure why that `implements` exists, is it for the case when reload happens?

Reply:
I thought it was a timing thing...I thought the `DeferredLog` needed to be initialized when the application started, if not its fine to remove it

Reply:
yeah, I've tested this locally and the logs are present if it is initialized in the constructor.